src/StoreApi/Authentication.php

```diff
@@ -55,4 +104,84 @@ protected function is_request_to_store_api() {
 		}
 		return 0 === strpos( $GLOBALS['wp']->query_vars['rest_route'], '/wc/store/' );
 	}
+
+	/**
+	 * Get current user IP Address.
+	 *
+	 * X_REAL_IP and CLIENT_IP are custom implementations designed to facilitate obtaining a user's ip through proxies, load balancers etc.
+	 *
+	 * _FORWARDED_FOR (XFF) request header is a de-facto standard header for identifying the originating IP address of a client connecting to a web server through a proxy server.
+	 * Note for X_FORWARDED_FOR, Proxy servers can send through this header like this: X-Forwarded-For: client1, proxy1, proxy2.
+	 * Make sure we always only send through the first IP in the list which should always be the client IP.
+	 * Documentation at https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For
+	 *
+	 * Forwarded request header contains information that may be added by reverse proxy servers (load balancers, CDNs, and so on).
+	 * Documentation at https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded
+	 * Full RFC at https://datatracker.ietf.org/doc/html/rfc7239
+	 *
+	 * @param boolean $proxy_support Enables/disables proxy support.
+	 * @return string|false
+	 */
+	protected static function get_ip_address( $proxy_support = false ) {
+
+		if ( ! $proxy_support ) {
+			if ( array_key_exists( 'REMOTE_ADDR', $_SERVER ) ) {
+				return self::validate_ip( sanitize_text_field( wp_unslash( $_SERVER['REMOTE_ADDR'] ) ) );
```

Comment:
If validation fails, do we want to return at this point, or continue to run through the other headers below?

Reply:
`REMOTE_ADDR` will always be set if we're not using CLI or the merchant has some odd server conf. Still, this makes sense, it is a loose end.

Maybe the perfect place to implement your suggestion of grouping requests where we cannot determine IP.

Reply:
We can just assign 0.0.0.0 for requests without an IP

Reply:
I added a return false for Mike's comment.

Currently, we return an error response if the IP can't be resolved, my line of thought was that something shady should happen if this is the case.

This is currently the only blocker merging this, @mikejolley @senadir do you think grouping those cases to share the same limits instead of returning an error response would be the best solution?

If so I'll make the change, if not things will be merged this way 👍🏼